container/dangerzone.py

```diff
@@ -181,64 +175,81 @@ def document_to_pixels(self) -> None:
-                ["pdftocairo", pdf_filename, "-png", "-singlefile", filename_base],
-                error_message="Conversion from PDF to PNG failed",
-                timeout_message=f"Error converting from PDF to PNG, pdftocairo timed out after {DEFAULT_TIMEOUT} seconds",
-            )
+        percentage_per_page = 45.0 / num_pages
+        page_base = "/tmp/page"
+        for page in range(1, num_pages + 1):
+            ppm_filename = f"{page_base}-{page}.ppm"
+            rgb_filename = f"{page_base}-{page}.rgb"
+            width_filename = f"{page_base}-{page}.width"
+            height_filename = f"{page_base}-{page}.height"
+            filename_base = f"{page_base}-{page}"
 
-            # Save the width and height
-            with Image.open(png_filename, "r") as im:
-                width, height = im.size
-            with open(width_filename, "w") as f:
-                f.write(str(width))
-            with open(height_filename, "w") as f:
-                f.write(str(height))
+            self.update_progress(f"Converting page {page}/{num_pages} to pixels")
 
-            # Convert to RGB pixels
+            # Convert to PPM, which is essentially an RGB format
             run_command(
                 [
-                    "gm",
-                    "convert",
-                    png_filename,
-                    "-depth",
-                    "8",
-                    f"rgb:{rgb_filename}",
+                    "pdftoppm",
+                    pdf_filename,
+                    "-f",
+                    str(page),
+                    "-l",
+                    str(page),
+                    "-singlefile",
+                    f"{page_base}-{page}",
                 ],
-                error_message="Conversion from PNG to RGB failed",
-                timeout_message=f"Error converting from PNG to pixels, convert timed out after {DEFAULT_TIMEOUT} seconds",
+                error_message="Conversion from PDF to PPM failed",
+                timeout_message=f"Error converting from PDF to PPM, pdftoppm timed out after {DEFAULT_TIMEOUT} seconds",
             )
 
-            # Delete the png
-            os.remove(png_filename)
+            with open(ppm_filename, "rb") as f:
+                # Read the header
+                header = f.readline().decode().strip()
+                if header != "P6":
+                    raise ValueError("Invalid PPM header")
+
+                # Save the width and height
+                dims = f.readline().decode().strip()
+                width, height = dims.split()
+                with open(width_filename, "w") as width_file:
+                    width_file.write(width)
+                with open(height_filename, "w") as height_file:
+                    height_file.write(height)
+
+                maxval = int(f.readline().decode().strip())
+                # Check that the depth is 8
+                if maxval != 255:
+                    raise ValueError("Invalid PPM depth")
```

Comment:
I don't expect this to be useful. Similarly to checking `P6` is the first header value, we check that the stated depth is 8 here. More of a sanity check than anything else. And since we need to get to the data that is after them, at least we use the opportunity to check if they're the values we expect.